specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json

```diff
@@ -0,0 +1,511 @@
+          "description": "A collection of the source event types which evaluate the playbook configuration set of rules.",
+          "items": {
+            "$ref": "#/definitions/PlaybookConfigurationSource"
+          }
+        },
+        "actions": {
+          "type": "array",
+          "description": "A collection of the actions which are triggered if all the configured rule set evaluation is true.",
+          "items": {
+            "$ref": "#/definitions/PlaybookConfigurationAction"
+          }
+        }
+      }
+    },
+    "PlaybookConfigurationMetadata": {
+      "type": "object",
+      "description": "The metadata of the playbook configuration resource.",
+      "properties": {
+        "createdDateTimeUtc": {
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time",
+          "description": "The playbook configuration creation date."
+        },
+        "createdBy": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The AAD object ID of the entity that created the playbook configuration."
+        },
+        "lastUpdatedDateTimeUtc": {
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time",
+          "description": "The playbook configuration last updated date."
+        },
+        "lastUpdatedBy": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The AAD object ID of the entity that last updated the playbook configuration."
+        }
+      }
+    },
+    "PlaybookConfigurationSource": {
+      "type": "object",
+      "description": "The source event types which evaluate the playbook configuration set of rules. For example - security alerts and recommended tasks.",
+      "properties": {
+        "eventSource": {
+          "type": "string",
+          "description": "A valid event source type."
+        },
+        "ruleSets": {
+          "type": "array",
+          "description": "A set of rules which evaluate upon event interception.",
+          "items": {
+            "$ref": "#/definitions/PlaybookConfigurationRuleSet"
+          }
+        }
+      }
+    },
+    "PlaybookConfigurationScope": {
+      "type": "object",
+      "description": "The subscription's resources scope",
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "The resources scope description."
+        },
+        "scopePath": {
+          "type": "string",
+          "description": "The resources scope path."
+        }
+      }
+    },
+    "PlaybookConfigurationAction": {
+      "type": "object",
+      "description": "The action that should be triggered.",
+      "properties": {
+        "resourceId": {
+          "type": "string",
+          "description": "The triggered resource id."
+        },
+        "resourceProviderNamespace": {
+          "type": "string",
+          "description": "The resource provider name."
+        },
```

Comment:
What is the relationship between `resourceId` and `resourceProviderNamespace`. Can the namespace not be inferred from the resource ID?

Reply:
This was changed. There will be future scenarios that need this information (for non-azure providers as well).

For now, this will include "logicAppResourceId", "resourceProviderNamespace" and "uri"

Reply:
Trying to jam settings specific to Azure and non-Azure RPs into one object will result in a confusing API because you have to guess which properties are valid for which scenario. 

My suggestion would be to add a kind property and use it as a discriminator when you get a 2nd scenario. Then, per value of the kind property, you can have a completely different schema. Take a look at how Microsoft.Blueprint RP is doing it with their artifacts resources. The good news is that you can add that later without breaking changes in the API with what you have so far.

---
In reply to: [281745248](https://github.com/Azure/azure-rest-api-specs/pull/5548#discussion_r281745248) [](ancestors = 281745248)

Reply:
Btw, my original point was that resourceProviderNamespace is redundant and can be inferred from the resourceId. In your examples, the Microsoft.Logic namespace is present in the resource ID. Why does the user have to specify it if we can just parse it out of the ID? Are there cases where the 2 namespaces are different?

---
In reply to: [282312303](https://github.com/Azure/azure-rest-api-specs/pull/5548#discussion_r282312303) [](ancestors = 282312303,281745248)

Reply:
The resource provider namespace can indeed be inferred however these will not always be Azure Resource Providers. We can change the name if this is too confusing but the meaning here is the "provider of the selected resource", not necessarily the "azure provider". If this is confusing, we can change the name.

Can you elaborate on the use of "kind" property? 
Code-wise, these are not the same objects, but this is added to apply different set of required access for each different ID type.

Reply:
Hi, we had additional thoughts on this - since we want several actions of different types and since we cant do this in the entire object, we added discriminator to the Action (actionType) and used this.